src/kakarot/instructions/system_operations.cairo

```diff
@@ -435,7 +427,9 @@ namespace CallHelper {
         ctx: model.ExecutionContext*, call_args: CallArgs
     ) {
         alloc_locals;
-        let (stack, popped) = Stack.pop_n(self=ctx.stack, n=6 + with_value);
+        // Note: We don't pop ret_offset and ret_size here but at the end of the sub context
+        // See finalize_calling_context
+        let (stack, popped) = Stack.pop_n(self=ctx.stack, n=4 + with_value);
```

Comment:
this is not the same stack

Reply:
so calling_ctx encounters CALL at pc X

we pop 4 from the stack instead of 6, 

we start the child execution context (different stack, different code)

we finalize the child context, come back up to calling_context, 
and then that calling_context still has 2 items of the stack to pop